backend/src/test/java/de/haiderzada/backend/controller/IngredientsItemControllerTest.java

```diff
@@ -0,0 +1,87 @@
+package de.haiderzada.backend.controller;
+
+import de.haiderzada.backend.model.Ingredient;
+import de.haiderzada.backend.model.Recipe;
+import de.haiderzada.backend.model.RecipeIngredient;
+import de.haiderzada.backend.repository.IngredientsRepository;
+import de.haiderzada.backend.repository.RecipesRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class IngredientsItemControllerTest {
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    private IngredientsRepository repository;
+
+    @Autowired
+    private RecipesRepository recipesRepository;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @BeforeEach
+    public void clearRepository() {
+        repository.deleteAll();
+    }
+
+    @Test
+    public void getIngredientsListfromDB() {
+        //GIVEN
+        repository.save(new Ingredient("1520", "Manti"));
+        repository.save(new Ingredient("1521", "Teigtasche"));
+
+        //WHEN
+        ResponseEntity<Ingredient[]> response = restTemplate.getForEntity("http://localhost:" + port + "/api/ingredients", Ingredient[].class);
+
+        //THEN
+        assertThat(response.getStatusCode(), is(HttpStatus.OK));
+        assertThat(response.getBody(), arrayContainingInAnyOrder(new Ingredient("1520", "Manti"), new Ingredient("1521", "Teigtasche")));
+
+    }
+
+    @Test
+    public void getfilteredRecipeListfromDB() {
+        //GIVEN
+        Ingredient ingredient = new Ingredient("9999","Banane");
+        Ingredient ingredient1 = new Ingredient("9998","Brot");
+        Ingredient ingredient2 = new Ingredient("9997","Nutella");
+
+        repository.save(ingredient);
+        repository.save(ingredient1);
+        repository.save(ingredient2);
```

Comment:
I'd save them just like in the test before for consistency

```suggestion
        repository.save(new Ingredient("9999","Banane"));
        repository.save(new Ingredient("9998","Brot"));
        repository.save(new Ingredient("9997","Nutella");
```

Reply:
When I save them directly, I can't use them in the next line again.